source/CompactRegular.lagda

```diff
@@ -0,0 +1,402 @@
+-- relation.
+well-inside : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → ⟨ F ⟩ → Ω 𝓤
+well-inside F U V = ∥ U ⋜₀[ F ] V ∥Ω
+
+infix 4 well-inside
+
+syntax well-inside F U V = U ⋜[ F ] V
+
+well-inside₀-implies-below : (F : frame 𝓤 𝓥 𝓦)
+                          → (U V : ⟨ F ⟩)
+                          → U ⋜₀[ F ] V
+                          → (U ≤[ poset-of F ] V) holds
+well-inside₀-implies-below F U V (W , c₁ , c₂) = connecting-lemma₂ F γ
+ where
+  _⊓_ = λ U V → U ∧[ F ] V
+
+  γ : U ≡ U ∧[ F ] V
+  γ = U                        ≡⟨ 𝟏-right-unit-of-∧ F U ⁻¹              ⟩
+      U ⊓ 𝟏[ F ]               ≡⟨ ap (U ⊓_) (c₂ ⁻¹)                     ⟩
+      U ⊓ (V ∨[ F ] W)         ≡⟨ binary-distributivity F U V W         ⟩
+      (U ⊓ V) ∨[ F ] (U ⊓ W)   ≡⟨ ap (λ - → binary-join F (U ⊓ V) -) c₁ ⟩
+      (U ⊓ V) ∨[ F ] 𝟎[ F ]    ≡⟨ 𝟎-left-unit-of-∨ F (U ⊓ V)            ⟩
+      U ⊓ V                    ∎
+
+\end{code}
+
+An open U in a frame F is *clopen* iff it is well-inside itself.
+
+\begin{code}
+
+is-clopen₀ : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → 𝓤 ̇ 
+is-clopen₀ F U = Σ W ꞉ ⟨ F ⟩ , (U ∧[ F ] W ≡ 𝟎[ F ]) × (U ∨[ F ] W ≡ 𝟏[ F ])
+
+is-clopen₀-is-prop : (F : frame 𝓤 𝓥 𝓦) → (U : ⟨ F ⟩) → is-prop (is-clopen₀ F U)
+is-clopen₀-is-prop F U (W₁ , p₁ , q₁) (W₂ , p₂ , q₂) = to-subtype-≡ β γ
+ where
+  P = poset-of F -- we refer to the underlying poset of F as P.
+
+  β : (W : ⟨ F ⟩) → is-prop ((U ∧[ F ] W ≡ 𝟎[ F ]) × (U ∨[ F ] W ≡ 𝟏[ F ]))
+  β W = ×-is-prop carrier-of-[ P ]-is-set carrier-of-[ P ]-is-set
+
+  γ : W₁ ≡ W₂
+  γ = W₁                                  ≡⟨ (𝟏-right-unit-of-∧ F W₁) ⁻¹       ⟩
+      W₁ ∧[ F ] 𝟏[ F ]                    ≡⟨ ap (λ - → meet-of F W₁ -) (q₂ ⁻¹) ⟩
+      W₁ ∧[ F ] (U ∨[ F ] W₂)             ≡⟨ binary-distributivity F W₁ U W₂   ⟩
+      (W₁ ∧[ F ] U) ∨[ F ] (W₁ ∧[ F ] W₂) ≡⟨ i                                 ⟩
+      (U ∧[ F ] W₁) ∨[ F ] (W₁ ∧[ F ] W₂) ≡⟨ ii                                ⟩
+      𝟎[ F ] ∨[ F ] (W₁ ∧[ F ] W₂)        ≡⟨ iii                               ⟩
+      (U ∧[ F ] W₂) ∨[ F ] (W₁ ∧[ F ] W₂) ≡⟨ iv                                ⟩
+      (W₂ ∧[ F ] U) ∨[ F ] (W₁ ∧[ F ] W₂) ≡⟨ v                                 ⟩
+      (W₂ ∧[ F ] U) ∨[ F ] (W₂ ∧[ F ] W₁) ≡⟨ vi                                ⟩
+      W₂ ∧[ F ] (U ∨[ F ] W₁)             ≡⟨ vii                               ⟩
+      W₂ ∧[ F ] 𝟏[ F ]                    ≡⟨ viii                              ⟩
+      W₂                                  ∎
+       where
+        i   = ap (λ - → - ∨[ F ] (W₁ ∧[ F ] W₂)) (∧[ F ]-is-commutative W₁ U)
+        ii  = ap (λ - → - ∨[ F ] (W₁ ∧[ F ] W₂)) p₁
+        iii = ap (λ - → - ∨[ F ] (W₁ ∧[ F ] W₂)) (p₂ ⁻¹)
+        iv  = ap (λ - → - ∨[ F ] (W₁ ∧[ F ] W₂)) (∧[ F ]-is-commutative U W₂)
+        v   = ap (λ - → (W₂ ∧[ F ] U) ∨[ F ] -) (∧[ F ]-is-commutative W₁ W₂)
+        vi  = binary-distributivity F W₂ U W₁ ⁻¹
+        vii = ap (λ - → W₂ ∧[ F ] -) q₁
+        viii = 𝟏-right-unit-of-∧ F W₂
```

Comment:
The indentation level of "=" is off by 1 in the last one.

Reply:
Done ✅